`azurerm_hdinsight_kafka_cluster`, `azurerm_hdinsight_spark_cluster`, `azurerm_hdinsight_interactive_query_cluster`, `azurerm_hdinsight_hbase_cluster`, `azurerm_hdinsight_hadoop_cluster` - add support for `disk_encryption_properties` property (#17351)
hashicorp/terraform-provider-azurerm: website/docs/r/hdinsight_spark_cluster.html.markdown

* `virtual_network_id` - (Optional) The ID of the Virtual Network where the Zookeeper Nodes should be provisioned within. Changing this forces a new resource to be created.

---

A `display_encryption_properties` block supports the following:

* `encryption_algorithm` - (Optional) This is an algorithm identifier for encryption. Possible values are `RSA1_5`, `RSA-OAEP`, `RSA-OAEP-256`.

> * `encryption_at_host` - (Optional) This is indicator to show whether resource disk encryption is enabled.

Reviewer: we put enabled on the end of bools, also should this be worded different
```suggestion
* `encrypted_at_host_enabled` - (Optional) This is indicator to show whether resource disk encryption is enabled.
```

> * `key_name` - (Optional) This is the name of key used for enabling disk encryption.
> 
> * `key_version` - (Optional) This is the specific version of the key used for enabling disk encryption.

Reviewer: these are key vault items?
```suggestion
* `key_vault_key_name` - (Optional) This is the name of key used for enabling disk encryption.

* `key_vault_key_version` - (Optional) This is the specific version of the key used for enabling disk encryption.
```

also could you not combine these into `key_vault_key_id` which takes the versionless or versioned id?

> * `msi_resource_id` - (Optional) This is the resource ID of Managed Identity used to access the key vault.

Reviewer: should this be something like
```suggestion
* `keyv_ault_managed_identity_id` - (Optional) This is the resource ID of Managed Identity used to access the key vault.
```

Author: Will change.

> * `vault_uri` - (Optional) this is the base key vault URI where the customers key is located.

Reviewer: and if we take the key vault key id above this can be removed?
